qbraid_qir/squin/visitor.py

```diff
@@ -0,0 +1,400 @@
+        globals (dict[str, Any] | None): The global variables to use. Defaults to None.
+        file (str | None): The file name for error reporting. Defaults to None.
+        lineno_offset (int): The line number offset for error reporting. Defaults to 0.
+        col_offset (int): The column number offset for error reporting. Defaults to 0.
+        compactify (bool): Whether to compactify the output. Defaults to True.
+
+    """
+
+    # Extract parameters from kwargs with defaults
+    kernel_name: str = kwargs.pop("kernel_name", "main")
+    dialects: ir.DialectGroup = kwargs.pop("dialects", kernel)
+    register_as_argument: bool = kwargs.pop("register_as_argument", False)
+    return_register: bool = kwargs.pop("return_register", False)
+    register_argument_name: str = kwargs.pop("register_argument_name", "q")
+    globals: dict[str, Any] | None = kwargs.pop("globals", None)
+    file: str | None = kwargs.pop("file", None)
+    lineno_offset: int = kwargs.pop("lineno_offset", 0)
+    col_offset: int = kwargs.pop("col_offset", 0)
+    compactify: bool = kwargs.pop("compactify", True)
+
+    # Raise error if unexpected kwargs are provided
+    if kwargs:
+        unexpected = ", ".join(f"'{k}'" for k in kwargs.keys())
+        raise TypeError(f"load() got unexpected keyword argument(s): {unexpected}")
+
+    # Validate input type at the start
+    if not isinstance(module, (str, pyqir.Module)):
+        raise InvalidSquinInput(f"Invalid input {type(module)}, expected 'str | pyqir.Module'")
+
+    # If module is a string, interpret as path to a file (.ll or .bc for QIR IR/bitcode)
+    # or as QIR IR text that can be parsed
+    if isinstance(module, str):
+        if os.path.exists(module):
+            _, ext = os.path.splitext(module)
+            if ext.lower() == ".ll":
+                # Load LLVM IR (text) file as a PyQIR module
+                with open(module, "r", encoding="utf-8") as f:
+                    ir_text = f.read()
+                module = pyqir.Module.from_ir(ir_text)
+            elif ext.lower() == ".bc":
+                # Load LLVM bitcode as a PyQIR module
+                with open(module, "rb") as f:
+                    bitcode_bytes = f.read()
+                module = pyqir.Module.from_bitcode(bitcode_bytes)
+            else:
+                raise InvalidFileException(f"Expected file extension .ll or .bc but got {ext!r}")
+        else:
+            # Try to parse string as QIR IR text
+            try:
+                module = pyqir.Module.from_ir(pyqir.Context(), module, name=kernel_name)
+            except Exception as exc:
+                raise InvalidSquinInput(
+                    f"Invalid input {type(module)}, expected 'str | pyqir.Module'. "
+                    f"String must be a valid file path (.ll or .bc) or valid QIR IR text."
+                ) from exc
+    elif isinstance(module, pyqir.Module):
+        # Already a valid PyQIR module, no conversion needed
+        pass
```

Comment:
This should be

```python
if not isintance(module pyqir.Module):
    raise Exception(...)
```

b/c at this point it should be a Module object. Right now the check is incomplete bc if the input module is not a string, it will just be let through and only raise an error once it reaches the `SquinVisitor` line